code/modules/surgery/organs/heart.dm

```diff
@@ -9,6 +9,47 @@
 	var/icon_base = "heart"
 	attack_verb = list("beat", "thumped")
 	var/beat = BEAT_NONE//is this mob having a heatbeat sound played? if so, which?
+	var/obj/item/assembly_holder/assembly = null
+	
+/obj/item/organ/heart/attackby(obj/item/I, mob/user, params)
+	if(!assembly && istype(I, /obj/item/assembly_holder))
+		. = 1
+		var/obj/item/assembly_holder/A = I
+		if(isigniter(A.a_left) == isigniter(A.a_right))
+			return
+		if(!user.transferItemToLoc(I, src))
+			return
+		assembly = A
+		A.master = src
+	return ..()
+	
+/obj/item/organ/heart/screwdriver_act(mob/user, obj/item/tool)
+	if(..())
+		return TRUE
+	if(assembly)
+		to_chat(user, "<span class='notice'>You remove [assembly]!</span>")
+		assembly.forceMove(drop_location())
+		assembly.master = null
+		assembly = null
+	
+/obj/item/organ/heart/IsAssemblyHolder()
+	return TRUE
+	
+/obj/item/organ/heart/receive_signal()
+	if(beating)
+		Stop()
+	else
+		Restart()
+		addtimer(CALLBACK(src, .proc/stop_if_unowned), 80) //The heart is a vunerable muscle.
+
+/obj/item/organ/heart/Crossed(atom/movable/AM)
```

Comment:
there's a signal, register for it

bruh

Reply:
COMSIG_MOVABLE_CROSSED